scim-core/src/main/java/org/apache/directory/scim/core/repository/Repository.java

```diff
@@ -60,14 +63,31 @@ public interface Repository<T extends ScimResource> {
   /**
    * Allows the SCIM server's REST implementation to update and existing
    * resource via a PUT to a valid end-point.
-   * 
-   * @param updateRequest The ScimResource to update and persist.
+   *
+   * @param id the identifier of the ScimResource to update and persist.
+   * @param version an optional version (usually used as an ETag) that be used to optimize update requests, may be compared against, the current {@code ScimResource.meta.version}.
```

Comment:
I'm a little confused by the `version <-> etag` relationship. 
are they are equivalent?
if etag an http concept then it makes sense to distinguish the two, but just curious if there was more to it than that

Reply:
also nit: that *can* be